plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py

```diff
@@ -183,10 +184,16 @@ def execute(self, **kwargs) -> Any:
             if is_runtime and issubclass(datatype, str):
                 final_batch_request["runtime_parameters"]["query"] = dataset
             elif is_runtime and issubclass(datatype, FlyteSchema):
-                final_batch_request["runtime_parameters"]["batch_data"] = dataset.open().all()
+                # if execution engine is SparkDF, transform the data to pyspark.sql.dataframe.DataFrame, else transform the data
+                # to the default pandas.dataframe
+                if selected_datasource[0]["execution_engine"]["class_name"] == "SparkDFExecutionEngine":
+                    final_batch_request["runtime_parameters"]["batch_data"] = dataset.open(
```

Comment:
nit: Could we move import to here? So current users don't necessarily have to install pyspark.
```suggestion
                if selected_datasource[0]["execution_engine"]["class_name"] == "SparkDFExecutionEngine":
                    import pyspark
                    final_batch_request["runtime_parameters"]["batch_data"] = dataset.open(
```

Reply:
good point, just moved pyspark.